backend/src/main/java/saci/application/UserEvaluationController.java

```
@@ -0,0 +1,205 @@
+    @ApiResponses(
+            value = {
+                @ApiResponse(
+                        responseCode = "200",
+                        description = "Score calculated successfully",
+                        content = {
+                            @Content(
+                                    mediaType = "application/json",
+                                    schema = @Schema(implementation = ScoreResponse.class))
+                        }),
+                @ApiResponse(
+                        responseCode = "404",
+                        description = "Role not found",
+                        content = @Content),
+                @ApiResponse(
+                        responseCode = "400",
+                        description = "Invalid request",
+                        content = @Content)
+            })
+    @PostMapping("/calculate-score")
+    public ResponseEntity<ScoreResponse> calculateScore(@Valid @RequestBody ScoreRequest request) {
+        Optional<Role> optionalRole = roleService.getRoleById(request.getRoleId());
+        if (!optionalRole.isPresent()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+        }
+        Role role = optionalRole.get();
+        List<Knowledge> knowledges = knowledgeService.getKnowledgesByRoleId(role.getId());
+
+        double totalWeightedScore = 0.0;
+        double totalWeight = 0.0;
+
+        for (Knowledge knowledge : knowledges) {
+            for (Answer answer : request.getAnswers()) {
+                if (knowledge.getId().equals(answer.getKnowledgeId())) {
+                    totalWeightedScore += answer.getAnswer() * knowledge.getWeight();
+                    totalWeight += knowledge.getWeight();
+                    break;
+                }
+            }
+        }
+
+        if (totalWeight == 0) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
```

Here you should throw a 500 error. 

Do you agree that, when totalWeight is 0, it´s not the user´s fault? So, it´s not a bad request, but a server error (the problem is in our side)